### PR TITLE
32/RLN-V1: Move to Draft

### PR DESCRIPTION
**Pull request by jimstir**
Update 32/RLN-V1 to move to draft.

### CODE REVIEW

File: vac/32/rln-v1.md [hunk 1]
```diff
@@ -22,19 +23,25 @@ Anonymity refers to the unlinkability of messages to their owner.
 
 RLN guarantees a messaging rate is enforced cryptographically while preserving the anonymity of the message owners.
 A wide range of applications can benefit from RLN and provide desirable security features.
-For example, an e-voting system can integrate RLN to contain the voting rate while protecting the voters-vote unlinkability.
-Another use case is to protect an anonymous messaging system against DDoS and spam attacks by containing messaging rate of users.
+For example, 
+an e-voting system can integrate RLN to contain the voting rate while protecting the voters-vote unlinkability.
+Another use case is to protect an anonymous messaging system against DDoS and 
+spam attacks by containing messaging rate of users.
```
Comment (rymnc) [inline vac/32/rln-v1.md:29]:
```suggestion
spam attacks by constraining messaging rate of users.
```

File: vac/32/rln-v1.md [hunk 2]
```diff
@@ -22,19 +23,25 @@ Anonymity refers to the unlinkability of messages to their owner.
...
 The users participate in the protocol by first registering to an application-defined group referred by the _membership group_.
 Registration to the group is mandatory for signaling in the application.
-After registration, group members can generate Zero-knowledge Proof of membership for their signals and can participate in the application.
-Usually, the membership requires a financial or social stake which
-is beneficial for the prevention of Sybil attacks and double-signaling.
-Group members are allowed to send one signal per external nullifier (an identifier that groups signals and can be thought of as a voting booth).
+After registration, group members can generate a zero-knowledge proof of membership for their signals and 
+can participate in the application.
+Usually, the membership requires a financial or 
+social stake which is beneficial for the prevention of Sybil attacks and double-signaling.
```
Comment (rymnc) [inline vac/32/rln-v1.md:42]:
```suggestion
social stake which is beneficial for the prevention of inclusion of Sybils within the _membership group_.
```

File: vac/32/rln-v1.md [hunk 3]
```diff
@@ -45,162 +52,180 @@ Generally the flow can be described by the following steps:
...
-
-## Registration
+### Registration
 
 Depending on the application requirements, the registration can be implemented in different ways, for example:
 - centralized registrations, by using a central server
 - decentralized registrations, by using a smart contract
 
-What is important is that the users' identity commitments (explained in section [User Identity](#user-identity)) are stored in a Merkle tree,
+What is important is that the users' identity commitments
```
Comment (rymnc) [inline vac/32/rln-v1.md:61]:
```suggestion
Tthe users' identity commitments 
```

File: vac/32/rln-v1.md [hunk 4]
```diff
@@ -45,162 +52,180 @@ Generally the flow can be described by the following steps:
...
-## Registration
+### Registration
 
 Depending on the application requirements, the registration can be implemented in different ways, for example:
 - centralized registrations, by using a central server
 - decentralized registrations, by using a smart contract
 
-What is important is that the users' identity commitments (explained in section [User Identity](#user-identity)) are stored in a Merkle tree,
+What is important is that the users' identity commitments 
+(explained in section [User Identity](#user-identity)) are stored in a Merkle tree,
```
Comment (rymnc) [inline vac/32/rln-v1.md:62]: can be in the same line as prev?

File: vac/32/rln-v1.md [hunk 5]
```diff
@@ -45,162 +52,180 @@ Generally the flow can be described by the following steps:
...
 and the users can obtain a Merkle proof proving that they are part of the group.
 
 Also depending on the application requirements,
 usually a financial or social stake is introduced.
+An example for financial stake is: 
 
-An example for financial stake is: For each registration a certain amount of ETH is required.
+For each registration a certain amount of ETH is required.
 An example for social stake is using InterRep as a registry -
```
Comment (rymnc) [inline vac/32/rln-v1.md:70]:
```suggestion
An example for social stake is using Interep as a registry -
```

Comment (rymnc) [reply vac/32/rln-v1.md:70]: and maybe a link: https://interep.link/

File: vac/32/rln-v1.md [hunk 6]
```diff
@@ -45,162 +52,180 @@ Generally the flow can be described by the following steps:
...
     identity_secret: [identity_nullifier, identity_trapdoor],
     identity_secret_hash: poseidonHash(identity_secret),
     identity_commitment: poseidonHash([identity_secret_hash])
 }
+
 ```
 
-For registration, the user needs to submit their `identity_commitment` (along with any additional registration requirements) to the registry.
-Upon registration, they should receive `leaf_index` value which represents their position in the Merkle tree.
+For registration, the user SHOULD submit their `identity_commitment`
```
Comment (rymnc) [inline vac/32/rln-v1.md:88]:
```suggestion
For registration, the user MUST submit their `identity_commitment` 
```

File: vac/32/rln-v1.md [hunk 7]
```diff
@@ -45,162 +52,180 @@ Generally the flow can be described by the following steps:
...
+to a centralised server.
 Along with their signal,
-they need to generate a ZK-Proof by using the circuit with the specification described above.
+they MUST generate a ZK-Proof by using the circuit with the specification described above.
 
 For generating a proof,
 the users need to obtain the required parameters or compute them themselves,
 depending on the application implementation and client libraries supported by the application.
-For example the users can store the membership Merkle tree on their end and
+For example, the users MAY store the membership Merkle tree on their end and
```
Comment (rymnc) [inline vac/32/rln-v1.md:105]:
```suggestion
For example, 
the users MAY store the membership Merkle tree on their end and
```

File: vac/32/rln-v1.md [hunk 8]
```diff
@@ -321,15 +355,13 @@ which together with the ZK-SNARK circuit enable the above mentioned features.
...
-
-### ZK Circuits specification
+#### ZK Circuits specification
 
 Anonymous signaling with a controlled rate limit is enabled by proving that the user is part of a group which has high barriers to entry (form of stake) and
 enabling secret reveal if more than 1 unique signal is produced per external nullifier.
-The membership part is implemented using membership [Merkle trees](https://en.wikipedia.org/wiki/Merkle_tree) and Merkle proofs,
-while the secret reveal part is enabled by using the Shamir's Secret Sharing scheme.
-Essentially the protocol requires the users to generate zero-knowledge proof to be able to send signals and participate in the application.
+The membership part is implemented using membership [Merkle trees](https://en.wikipedia.org/wiki/Merkle_tree) and Merkle proofs, while the secret reveal part is enabled by using the Shamir's Secret Sharing scheme.
```
Comment (rymnc) [inline vac/32/rln-v1.md:362]:
```suggestion
The membership part is implemented using membership [Merkle trees](https://en.wikipedia.org/wiki/Merkle_tree) and Merkle proofs, 
while the secret reveal part is enabled by using the Shamir's Secret Sharing scheme.
```

File: vac/32/rln-v1.md [hunk 9]
```diff
@@ -321,15 +355,13 @@ which together with the ZK-SNARK circuit enable the above mentioned features.
...
+Essentially the protocol requires the users to generate zero-knowledge proof to be able to send signals and
```
Comment (rymnc) [inline vac/32/rln-v1.md:363]: there are usages of `zero-knowledge proof` and `zk-proof` interchangeably, either we should mention this up top, or use one throughout

**Review by rymnc — APPROVED** [review]
LGTM, minor nits inline :) Thanks!

File: vac/32/rln-v1.md (hunk 9 above)
Comment (jimstir) [reply vac/32/rln-v1.md:363]: I replaced `zk-proof` with `zero-knowledge proof` within the spec. Expect for `zk_proof` value, I'm not sure if this should be changed.

File: vac/32/rln-v1.md (hunk 4 above)
Comment (jimstir) [reply vac/32/rln-v1.md:62]: Im not sure what you mean.

**Review by rymnc — APPROVED** [review]
LGTM

**Pull request merged**